fetch-users.test.ts

```diff
@@ -0,0 +1,52 @@
+import { beforeEach, describe, it, vi, expect } from "vitest";
+import fetch, { type Response } from "node-fetch";
+import { fetchUsers } from "./fetch-users.js";
+
+vi.mock("node-fetch");
+
+describe("fetchUsers", () => {
+    beforeEach(() => {
+        // Reset the mock before each test run to ensure number of calls is reset back to 0 after each test
+        vi.mocked(fetch).mockReset();
+    });
+
+    it("it returns an empty array when the webhook is empty", async () => {
+        vi.mocked(fetch).mockResolvedValue({
+            json: async () => [],
+        } as Response);
+
+        const userArray: User[] = await fetchUsers();
+
+        expect(userArray.length).toEqual(0);
+        expect(userArray).toEqual([]);
+        expect(fetch).toHaveBeenCalledTimes(1);
+    });
+
+    it("it returns an empty array when the webhook contains users", async () => {
```

Comment:
This test title is slightly at odds with what the test actually does - perhaps it could be tweaked slightly?